### ISSUE TITLE
Add support for Version field optimistic currency to the NHibernate saga persister

### ISSUE THREAD

**Issue by andreasohlund** [Particular/NServiceBus#1084]
When v3 was released the support for a Version field to handle optimistic concurrency was accidentally removed. Since v3 runs with isolation level == serializable by default this has causes issues for the users. In v4 we default to readcommitted so the needs to be fixed. Issue #649 introduces support for rowlevel concurrency but since this can cause performance issue we need to add the support for a version field back. 

In short:
- We default to row level concurrency (like we do it at the moment)
- Introduces a [Version] attribute that the user can put on their version property.
- If the [Version] is found, enable field level concurrency accordingly
- Add a config option to turn off optimistic concurrent completely for users that wants to take full control

http://tech.groups.yahoo.com/group/nservicebus/message/18409
